docs: add warning about rem unit inheritance in Shadow Root UI (#2330)
wxt-dev/wxt: docs/guide/essentials/content-scripts.md

Reviewer: Use those tags instead of `:::key`:
https://github.com/orgs/community/discussions/16925

And see this file on GH, to be sure it looks and works well

Reviewer: You can't set text next to [!WARNING[.
Like smah mentioned here:
https://github.com/wxt-dev/wxt/pull/2336#issuecomment-4394097768

Let's see how it looks on github.
https://github.com/Aniket-lodh/wxt/blob/7a94f07344d67c6e5321c73aabca0ebaf3020a1c/docs/guide/essentials/content-scripts.md

Let's check it next time, if everything is OK, before you'll request any review.
You're almost there 😄

Reviewer 2: I've switched it to `:::warning`

- [react-content-script-ui](https://github.com/wxt-dev/examples/tree/main/examples/react-content-script-ui)
- [tailwindcss](https://github.com/wxt-dev/examples/tree/main/examples/tailwindcss)

> :::warning `rem` Units Are Not Fully Isolated

Reviewer: This also isn't work.

Let's watch, how it looks overall:
https://github.com/Aniket-lodh/wxt/edit/docs/shadow-root-rem-inheritance/docs/guide/essentials/content-scripts.md

> While `createShadowRootUi` isolates most CSS, **`rem` units are not isolated** inside the Shadow DOM. This is because `rem` is relative to the root `<html>` element's `font-size`, which lives _outside_ the Shadow Root. WXT v0.20+ applies `all: initial` to reset inherited properties like `visibility`, `color`, and `font-size`, but this does **not** affect how `rem` values are computed.

Reviewer: ```suggestion
While `createShadowRootUi` isolates most CSS, **`rem` units aren't isolated** inside the Shadow DOM. That's because `rem` is relative to the root `<html>` element's `font-size`, which lives _outside_ the Shadow Root.
Property `all: initial` **doesn't** affect how `rem` values.
```

If the host website sets a custom `font-size` on the `<html>` element (e.g., Reddit uses `font-size: 10px`, some sites use `62.5%`), all `rem`-based styles inside your Shadow Root UI will scale incorrectly — making your extension UI appear too large or too small depending on the website.

> This affects any CSS framework that uses `rem` units by default, including **Tailwind CSS**.

Reviewer: ```suggestion

This affects any CSS framework that uses `rem` units by default, including **Tailwind CSS**.
```

**Fix: Convert `rem` to `px` at build time**

Use [`postcss-rem-to-responsive-pixel`](https://www.npmjs.com/package/postcss-rem-to-responsive-pixel) to automatically convert `rem` units to `px` during the build, eliminating the dependency on the host page's root font-size.

1. Install the package:

   ```sh

>    npm i -D postcss-rem-to-responsive-pixel

Reviewer: ```suggestion
   bun i -D postcss-rem-to-responsive-pixel
```

   ```

2. Configure your PostCSS config:

>    :::code-group

Reviewer: Use `<CodeGroup>...</CodeGroup>` instead, because your solution doesn't work well on GH

   ```js [postcss.config.mjs (ESM)]
   import tailwindcss from '@tailwindcss/postcss';
   import postcssRemToResponsivePx from 'postcss-rem-to-responsive-pixel';

   export default {
     plugins: [
       tailwindcss({}),
       postcssRemToResponsivePx({
         rootValue: 16,
         propList: ['*'],
         transformUnit: 'px',
       }),
     ],
   };
   ```

   ```js [postcss.config.cjs (CJS)]
   module.exports = {
     plugins: [
       require('@tailwindcss/postcss')({}),
       require('postcss-rem-to-responsive-pixel')({
         rootValue: 16,
         propList: ['*'],
         transformUnit: 'px',
       }),
     ],
   };
   ```

   :::

   > Set `rootValue` to `16` (the browser default) so that `1rem` converts to `16px`. Adjust if your design system uses a different base.
   >

>    > If you are not using Tailwind CSS, remove the `tailwindcss` plugin line and keep only `postcssRemToResponsivePx`.

Reviewer: ```suggestion
   ```js [postcss.config.mjs (ESM)]
   import postcssRemToResponsivePx from 'postcss-rem-to-responsive-pixel';

   export default {
     plugins: [
       postcssRemToResponsivePx({
         rootValue: 16,
         propList: ['*'],
         transformUnit: 'px',
       }),
     ],
   };
   ```

   ```js [postcss.config.cjs (CJS)]
   module.exports = {
     plugins: [
       require('postcss-rem-to-responsive-pixel')({
         rootValue: 16,
         propList: ['*'],
         transformUnit: 'px',
       }),
     ],
   };
   ```

   :::

   > Set `rootValue` to `16` (the browser default) so that `1rem` converts to `16px`. Adjust if your design system uses a different base.
```

I think it isn't necessary to that docs, unnecessary distractor.
Only `postcssRemToResponsivePx` is crucial here.
